src/barril/units/_scalar.py

```diff
@@ -254,7 +254,8 @@ def AlmostEqual(self, other, precision):
         )
 
     def __hash__(self, *args, **kwargs):
-        return hash((self._value, self._quantity))
+        default_unit = self._unit_database.GetDefaultUnit(self._quantity._category)
```

Comment:
IMO `__eq__` should be using the same logic when selecting the unit used for comparison.
Does this work with composed units?

Reply:
You mean do the comparison always using the `DefaultUnit`?

Reply:
Yes.

Reply:
Also because equal objects must have the same hash.

Reply:
Ok agreed, I've refactored it and tested against composed units. Thanks for notice that.

Reply:
Take a look at the refactored version